### PR TITLE
Add signer utilities

### PR DESCRIPTION
**Pull request by loosebazooka**
Signed-off-by: Appu Goundan <appu@google.com>

### CODE REVIEW

File: src/main/java/dev/sigstore/encryption/signers/Signer.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import java.security.PublicKey;
+import java.security.SignatureException;
+
+/** A signing helper that wraps common signing operations for use within this library. */
+public interface Signer {
+
+  /** Return the public key associated with this signer. */
+  PublicKey getPublicKey();
+
+  /** Sign using the singers algorithm with sha256 digest. */
```
Comment (patflynn) [inline src/main/java/dev/sigstore/encryption/signers/Signer.java:30]: singers?

**Review by patflynn — DISMISSED** [review]
Seems like a nice improvement.

File: src/main/java/dev/sigstore/encryption/signers/Signer.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  byte[] sign(String content, Charset charset)
+      throws NoSuchAlgorithmException, InvalidKeyException, SignatureException;
+
+  /** Sign using the singers algorithm with sha256 digest. */
```
[the remark this answers is not in the document]
Comment (loosebazooka) [reply src/main/java/dev/sigstore/encryption/signers/Signer.java:34]: oh woops. singer's

[the remark this answers is not in the document]
Comment (loosebazooka) [reply src/main/java/dev/sigstore/encryption/signers/Signer.java:34]: signer's*

**Review by patflynn — APPROVED** [review]

**Pull request merged**